packages/redirect/src/class-redirect.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Jetpack Redirect package.
+ *
+ * @package  automattic/jetpack-redirect
+ */
+
+namespace Automattic\Jetpack;
+
+/**
+ * Class Redirect
+ */
+class Redirect {
+	/**
+	 * Constructor.
+	 *
+	 * Static-only class, so nothing here.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Strip http:// or https:// from a url, replaces forward slash with ::,
+	 * so we can bring them directly to their site in calypso.
+	 *
+	 * @param string $url the full URL.
+	 * @return string $url without the guff
+	 */
+	private static function build_raw_urls( $url ) {
+		$strip_http = '/.*?:\/\//i';
+		$url        = preg_replace( $strip_http, '', $url );
+		$url        = str_replace( '/', '::', $url );
+		return $url;
+	}
+
+	/**
+	 * Builds and returns an URL using the jetpack.com/redirect service
+	 *
+	 * Note to WP.com: Changes to this method must be synced to wpcom
+	 *
+	 * @param string       $source The URL handler registered in the server.
+	 * @param array|string $args {
+	 *    Optional. Additional arguments to build the url.
+	 *
+	 *    @type string $site URL of the site; Default is current site.
+	 *    @type string $path Additional path to be appended to the URL.
+	 *    @type string $query Query parameters to be added to the URL.
+	 *    @type string $anchor Anchor to be added to the URL.
+	 * }
+	 *
+	 * @return string The built URL
+	 */
+	public static function get_url( $source, $args = array() ) {
+
+		$url           = 'https://jetpack.com/redirect';
```

Comment:
I had a thought where we might want to make this filterable, but can't really think of any use-cases where someone would want to mimic this on their own server redirect. We can make it filterable later if desired.

Reply:
wouldn't it be a security riks? A plugin that hijacks all links from jetpack...?

Reply:
A plugin could do a lot worse than that ;)